src/main/kotlin/nl/avisi/structurizr/site/generatr/site/PlantUmlExporter.kt

```diff
@@ -0,0 +1,165 @@
+        writer.writeLine("skinparam preserveAspectRatio meet")
+    }
+
+    fun writeElement(
+        view: ModelView?,
+        element: Element?,
+        writer: IndentingWriter?,
+        writeElementFn: (view: ModelView?, element: Element?, writer: IndentingWriter?) -> Unit
+    ) {
+        val url = when {
+            needsLinkToSoftwareSystem(element, view) -> getUrlToSoftwareSystem(element)
+            needsLinkToContainerViews(element, view) -> getUrlToContainerViews(element)
+            needsLinkToComponentViews(element, view) -> getUrlToComponentViews(element)
+            else -> null
+        }
+
+        if (url != null)
+            writeElementWithCustomUrl(element, url, view, writer, writeElementFn)
+        else
+            writeElementFn(view, element, writer)
+    }
+
+    private fun needsLinkToSoftwareSystem(element: Element?, view: ModelView?) =
+        element is SoftwareSystem && workspace.includedSoftwareSystems.contains(element) && element != view?.softwareSystem
+
+    private fun getUrlToSoftwareSystem(element: Element?): String {
+        val path = "/${element?.name?.normalize()}/context/".asUrlToDirectory(url)
+        return "$TEMP_URI$path"
+    }
+
+    private fun needsLinkToContainerViews(element: Element?, view: ModelView?) =
+        element is SoftwareSystem && workspace.includedSoftwareSystems.contains(element) && element == view?.softwareSystem && element.hasContainers
+
+    private fun getUrlToContainerViews(element: Element?): String {
+        val path = "/${element?.name?.normalize()}/container/".asUrlToDirectory(url)
+        return "$TEMP_URI$path"
+    }
+
+    private fun needsLinkToComponentViews(element: Element?, view: ModelView?) =
+        element is Container && element.hasComponents && view !is ComponentView
+
+    private fun getUrlToComponentViews(element: Element?): String {
+        val path = "/${element?.parent?.name?.normalize()}/component/".asUrlToDirectory(url)
+        return "$TEMP_URI$path"
+    }
+
+    private fun writeElementWithCustomUrl(
+        element: Element?,
+        url: String?,
+        view: ModelView?,
+        writer: IndentingWriter?,
+        writeElementFn: (view: ModelView?, element: Element?, writer: IndentingWriter?) -> Unit
+    ) {
+        element?.url = url
+        writeModifiedElement(view, element, writer, writeElementFn)
+        element?.url = null
+    }
+
+    private fun writeModifiedElement(
+        view: ModelView?,
+        element: Element?,
+        writer: IndentingWriter?,
+        writeElement: (view: ModelView?, element: Element?, writer: IndentingWriter?) -> Unit
+    ) = IndentingWriter().let {
+        writeElement(view, element, it)
+        it.toString()
+            .replace(TEMP_URI, "")
+            .split(System.lineSeparator())
+            .forEach { line -> writer?.writeLine(line) }
+    }
+}
+
+class C4PlantUmlExporterWithElementLinks(workspace: Workspace, url: String) : C4PlantUMLExporter() {
```

Comment:
This could be private as well, but then the test setup will differ slightly. Probably i good idea to change this.